test/framework/src/main/java/org/elasticsearch/test/AbstractXContentTestCase.java

```diff
@@ -54,21 +54,16 @@
         for (int runs = 0; runs < numberOfTestRuns; runs++) {
             T testInstance = instanceSupplier.get();
             XContentType xContentType = randomFrom(XContentType.values());
-            BytesReference xContent = XContentHelper.toXContent(testInstance, xContentType, ToXContent.EMPTY_PARAMS, false);
-            BytesReference withRandomFields;
-            if (supportsUnknownFields) {
-                // we add a few random fields to check that parser is lenient on new fields
-                withRandomFields = XContentTestUtils.insertRandomFields(xContentType, xContent, randomFieldsExcludeFilter, random());
-            } else {
-                withRandomFields = xContent;
-            }
-            XContentParser parserWithRandonFields = createParserFunction.apply(XContentFactory.xContent(xContentType), withRandomFields);
-            BytesReference shuffledContent = BytesReference.bytes(shuffleXContent(parserWithRandonFields, false, shuffleFieldsExceptions));
+            BytesReference shuffledContent = insertRandomFieldsAndShuffle(testInstance, xContentType, supportsUnknownFields,
+                    shuffleFieldsExceptions, randomFieldsExcludeFilter, createParserFunction, toXContentParams);
             XContentParser parser = createParserFunction.apply(XContentFactory.xContent(xContentType), shuffledContent);
             T parsed = parseFunction.apply(parser);
             assertEqualsConsumer.accept(testInstance, parsed);
             if (assertToXContentEquivalence) {
-                assertToXContentEquivalent(xContent, XContentHelper.toXContent(parsed, xContentType, false), xContentType);
+                assertToXContentEquivalent(
+                        XContentHelper.toXContent(testInstance, xContentType, false),
```

Comment:
I have a hunch that this will again need to be changed using the variant that takes the `toXContentParams` argument. Will dig.